src/lib.rs

```diff
@@ -855,6 +856,20 @@ impl<T, C: MemConfig> IntoIterator for SegVec<T, C> {
     }
 }
 
+/// Creates an new [`SegVec`][crate::SegVec] from a [`Slice`][crate::Slice].
+impl<T: Clone, C: MemConfig> From<Slice<'_, T>> for SegVec<T, C> {
+    fn from(slice: Slice<'_, T>) -> Self {
+        slice.iter().cloned().collect()
+    }
+}
+
+/// Creates an new [`SegVec`][crate::SegVec] from a reference to [`Slice`][crate::Slice].
+impl<T: Clone, C: MemConfig> From<&Slice<'_, T>> for SegVec<T, C> {
+    fn from(slice: &Slice<'_, T>) -> Self {
+        slice.iter().cloned().collect()
+    }
+}
```

Comment:
Is there any specific reason for implementing `From` directly for `Slice`, `&Slice`? It seems like these both just forward to the `FromIterator` implementation via `collect`, so we could just support `FromIterator` for iterators over reference types when the target type is `Clone`, something like:

```rust
impl<'a, T: Clone + 'a, C: MemConfig> Extend<&'a T> for SegVec<T, C> {
    fn extend<I: IntoIterator<Item = &'a T>>(&mut self, iter: I) {
        <Self as Extend<T>>::extend(self, iter.into_iter().cloned())
    }
}
impl<'a, T: 'a + Clone, C: MemConfig> FromIterator<&'a T> for SegVec<T, C> {
    fn from_iter<I: IntoIterator<Item = &'a T>>(iter: I) -> Self {
        let mut v = SegVec::new();
        v.extend(iter);
        v
    }
}
```

There's already an `Extend` implementation for `&'a T` when `T: Copy`, so we could just change that to be `T: Clone`, since `T: Copy` requires `T: Clone` anyway and the derived `Clone` implementation for `T: Copy` should, from what I understand, just be a memcopy.